golem-rib/src/type_inference/mod.rs

```diff
@@ -259,13 +262,6 @@ mod type_inference_tests {
 
     #[test]
     fn test_inference_inline_type_annotation_7() {
-        // ok(1) isn't automatically inferred since no further hints are given
-        let mut invalid_rib_expr = Expr::from_text(r#"ok(1)"#).unwrap();
-
-        let result = invalid_rib_expr.infer_types(&FunctionTypeRegistry::empty(), &vec![]);
-
-        assert!(result.is_err());
```

Comment:
These will get inferred with this change. They are not compilation errors anymore.